### ISSUE TITLE
After version 3.3.7.0, the video from the camera goes with a delay of 5 -10 seconds

### ISSUE THREAD

**Issue by DivanX10** [doitandbedone/ispyagentdvr-docker#298]
**Affected Version**
Agent v3.3.7.0

**Describe the bug**
I use the docker doitandbedone/ispyagentdvr 3.3.7.0 image. With this version, the video stream from the camera goes fast and without delays, like with webrtc . As soon as I put the image fresher, higher than version 3.8.0, the video starts to slow down, it comes with a delay of 5-10 seconds. At first I couldn't understand what the problem was. I started downgrading versions from the latest 3.5.2.0 and when I got to version 3.3.7.0. the video works without delays in real time, like webrtc. I use the same settings, upload the same configuration files "agentDVR.xml" and "agentConfig.xml" through system recovery.

Camera Eufy Indoor Cam 2K Pan & Tilt
Docker works on NAS Synology
Hardware Configuration NAS Synology CPU: Core i7 3770 + RAM: 16 gb

![image](https://user-images.githubusercontent.com/64090632/128530531-2e0c2f07-e4b5-4077-8f63-df9187add3c6.png)

**Comment by doitandbedone** [comment]
Interesting, would you mind attaching the corresponding logs for each of
the versions? That would let us know what may be happening here.

Thank you!

**Comment by DivanX10** [comment]
And where can I get these magazines? Are you talking about these logs (screenshot below)? You can check for yourself. Take any version of agent dvr lower than 3.3.7.0 and any version higher than 3.3.8.0 Install Home Assistant. Add the Agent DVR integration and see what the video stream will be. Agent dvr version lower than 3.3.7.0 has a video stream without delays, and agent dvr version higher than 3.3.8.0 already has delays of 5-10 seconds.

![image](https://user-images.githubusercontent.com/64090632/128556492-e7bdc7d5-0c75-4d0c-a915-ff28ce30211d.png)

**Comment by ispysoftware** [comment]
v3.3.8.0 added max delay and reorder queue size to the video source
settings -> ffmpeg settings - this is needed for streams where the video/
audio is sent in large chunks and needs to be re-ordered for non stuttering
playback. You can set both to zero which should be the same as <3.3.8.0

**Comment by DivanX10** [comment]
agent dvr of version 3.5.2.0
I played around with the settings in ffmpegsettings, and it was not possible to achieve a real-time stream without delays
![image](https://user-images.githubusercontent.com/64090632/128596400-158bedfb-b534-4cec-b373-6bd3119dda7c.png)

agent dvr of version  3.3.7.0
Everything is identical here, but the video stream goes in real time without delays
![image](https://user-images.githubusercontent.com/64090632/128596604-f710d0bf-c825-40bd-9009-de47dcb47865.png)

**Comment by ispysoftware** [comment]
that's the only relevant change between those 2 versions - did you disable/
enable the camera after changing it?

**Comment by DivanX10** [comment]
Yes, when I change the settings, I click restart Agent DVR. I also tried to reboot the camera. What is important is that when I switch from version 3.5.2.0 to 3.3.7.0, I don't need to reboot the camera there, it's enough to restart the Agent DVR and the video goes in real time. Apparently, I will then sit on version 3.3.7.0, since this is the latest version where the video is broadcast without delays and in real time.

**Comment by ispysoftware** [comment]
Ah I think i see the issue, looks like some cameras are producing delayed
timestamps which is affecting the new playback throttling code - i'll
update it to bypass that stuff if max delay is set to 0. Will be in next
update.

**Comment by DivanX10** [comment]
Thanks. I will wait. If it is possible, then please do not close this topic yet, so that later I can test the new version

**Comment by doitandbedone** [comment]
We usually check back before closing issues.

**Comment by ispysoftware** [comment]
uploaded an update to the site, give that a go, set maxdelay to 0

**Comment by DivanX10** [comment]
I took it into account. Until I see the last image. Now there is a version 3.5.2.0. Maybe it hasn't loaded yet, you need to wait for some time?

![image](https://user-images.githubusercontent.com/64090632/130220085-a56b3d1f-a182-48fc-ad2d-06354a1449c9.png)

**Comment by doitandbedone** [comment]
Yes, give it some time. It does runs automatically throughout the day, I'll
force a run manually though.

**Comment by doitandbedone** [comment]
Ok, jobs kicked off, you should receive the update in a few minutes. Look for latest or 3.5.4.0 if updating manually

**Comment by DivanX10** [comment]
Until the image appeared here https://registry.hub.docker.com/r/doitandbedone/ispyagentdvr/tags?page=1&ordering=last_updated, and the links can't be installed in docker synology. I'll wait for it to appear in the docker

**Comment by DivanX10** [comment]
I found the container. It turns out that it has the signature of the latest and does not have a version number. Installed it. Now it works as fast as version 3.3.7.0, but we need to test more time. I think that this branch can be closed. Thank you very much for your work.

![image](https://user-images.githubusercontent.com/64090632/130319339-22cb806c-8cfa-424f-8b45-044867ca6677.png)
